### PR TITLE
AzureMonitor: Fix filter query for empty queryType

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

I noticed that the default query was being executed in alerting when createing a new alert. For some reason, this does not happen in a normal dashboard (I assume the query gets "migrated" in a normal dashboard, which adds a queryType, but that does not happen in Alerting)

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Related to https://github.com/grafana/support-escalations/issues/4577

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.test.ts [hunk 1]
```diff
@@ -34,4 +34,17 @@ describe('Azure Monitor Datasource', () => {
...
     });
   });
+
+  it('should not filter a valid query', () => {
+    const ds = new Datasource(createMockInstanceSetttings());
+    const query = createMockQuery();
+    expect(ds.filterQuery(query)).toBe(true);
+  });
+
+  it('should not filter query with no query type', () => {
```
Comment (aangelisc) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.test.ts:44]: I think this should be `should filter query with no query type`?

Comment (aangelisc) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.test.ts:44]: It also confuses me that `filterQuery` returns true for a query that *shouldn't* be filtered. That feels counter- intuitive to me?

Comment (andresmgot) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.test.ts:44]: the syntax is complex :sweat_smile: but it's the default behavior in JS, executions that return `true` are kept:

```
Welcome to Node.js v16.17.0.
Type ".help" for more information.
> [true, false].filter(e => e)
[ true ]
> 
```

Comment (aangelisc) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.test.ts]: It's always baffled me that this was the default behaviour 😂

**Review by aangelisc — APPROVED** [review]

**Pull request merged**